include/ssl_key_handler.hpp

```diff
@@ -556,7 +563,7 @@ inline std::optional<boost::asio::ssl::context> getSSLClientContext()
         BMCWEB_LOG_ERROR("SSL context set_verify_mode failed");
         return std::nullopt;
     }
-
+#endif
```

Comment:
1) Instead using `#if 0`, I think it is better to add an option (`bmcweb-peer-cert-verify`) and make it as `disabled` by default?

After that, upstream it (with enabling it).

Wouldn't this be a better way to handle this?

2) 1060, this part of code is not disabled.  
     Why would need to be disabled in 1110?

https://github.com/ibm-openbmc/bmcweb/blob/a33fa9a4d1528126ba92b9290915811ef1132201/include/ssl_key_handler.hpp#L497

Reply:
we are already doing the same for 1050 and 1060 code base and as per our discussions with HMC team, looks like HMC can't make changes in their certificate flows for p11.
we have already discussed this upstream, for now we should disable  server verification

Reply:
> 1. Instead using `#if 0`, I think it is better to add an option (`bmcweb-peer-cert-verify`) and make it as `disabled` by default?
> 
> After that, upstream it (with enabling it).
> 
> Wouldn't this be a better way to handle this?
> 
> 2. 1060, this part of code is not disabled.
>    Why would need to be disabled in 1110?
> 
> https://github.com/ibm-openbmc/bmcweb/blob/a33fa9a4d1528126ba92b9290915811ef1132201/include/ssl_key_handler.hpp#L497

1060 does not use this method at all

Reply:
Till now HMC certificate verification was never supported on BMC systems, never heard of any security concern as such till now on any , especially this server certificate verification part which hit in redfish event flow

Reply:
" never heard of any security concern" does not mean that there is no vulnerability exists. It is a potential threat and now any one can subscribe to redfish events which probably has some secure data. Yes it is an inconvenience during development, but we can solve it by uploading self signed CA to bmc.

Reply:
> we are already doing the same for 1050 and 1060 code base and as per our discussions with HMC team, looks like HMC can't make changes in their certificate flows for p11. we have already discussed this upstream, for now we should disable server verification

Yes ,we have already discussed about this in our scrum. I pointed out this in this commit to make the note on where the actual fix supposed to go.

Reply:
My previous comment is mainly about the code management  so that we can potentially avoid any more downstream-only patches in the future by doing;
- upstream bmcweb - Add a meson option to enable/disable this part
- upstream bmcweb - make it as enabled (`meson_options.txt`)
- downstream openbmc - make it as disabled (`bmcweb_%.bbappend`)

In this way, as long as this part needs to be different between upstream and downstream, we can keep using the same code in both upstream & downstream.